pkg/skaffold/kubernetes/sync.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/sync"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type KubectlSyncer struct{}
+
+func (k *KubectlSyncer) Sync(s *sync.Item) error {
+	if err := perform(s.Image, s.Copy, copyFileFn); err != nil {
+		return errors.Wrap(err, "copying files")
+	}
+	if err := perform(s.Image, s.Delete, deleteFileFn); err != nil {
+		return errors.Wrap(err, "deleting files")
+	}
+	return nil
+}
+
+func deleteFileFn(pod v1.Pod, container v1.Container, src, dst string) *exec.Cmd {
+	return exec.Command("kubectl", "exec", pod.Name, "--namespace", pod.Namespace, "-c", container.Name, "--", "rm", "-rf", dst)
+}
+
+func copyFileFn(pod v1.Pod, container v1.Container, src, dst string) *exec.Cmd {
+	return exec.Command("kubectl", "cp", src, fmt.Sprintf("%s/%s:%s", pod.Namespace, pod.Name, dst), "-c", container.Name)
+}
```

Comment:
A debug log would be useful here.

Reply:
We get a debug log with util.RunCmd. Does that work?

Reply:
yes should be good